src/bridge/cockpitdisksamples.c

```diff
@@ -148,3 +153,124 @@ cockpit_disk_samples (CockpitSamples *samples)
   g_strfreev (lines);
   g_free (contents);
 }
+
+static gboolean
+read_file (const gchar *path,
+           gchar **contents)
+{
+  g_autoptr(GError) error = NULL;
+  if (!g_file_get_contents (path, contents, NULL, &error))
+    {
+      // do not log following erros:
+      // file does not exist: process died and can no longer be sampled
+      // access denied: bridge is not privileged and can't sample all processes
+      // sometimes ERROR_FAILED is returned with message: "No such process"
+      if (!(g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT) ||
+          g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_ACCES) ||
+          (g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_FAILED) && g_strrstr(error->message, "No such process"))))
+        {
+          g_warning ("Domain: %s, code: %d, message: %s", g_quark_to_string (error->domain), error->code, error->message);
+        }
+
+      return FALSE;
+    }
+
+  return TRUE;
+}
+
+static void
```

Comment:
I'm not a fan of how you read the key (the cgroup name) from this function, but the values (the read/write figures) from the other function.  It also feels weird that you pass the allocated struct across the function boundary, expecting it to be consumed — that's highly irregular.  It also feels weird that, if you find the item already in the hash table, you modify the new item and replace the old item, instead of modifying the existing item, in-place.

I feel like it would be more natural to have some lower-level functions exclusively to deal with the hash table, something like:

```C
void
table_add_values (GHashTable *table,
                  const gchar *cgroup,
                  guint64 read,
                  guint64 write)
{
  // create or update the hash table entry for this cgroup
}
```

and then you could get the key and values from a single function and call this function.